storage/memory/log_storage.go

```diff
@@ -342,6 +342,22 @@ func (t *logTreeTX) GetActiveLogIDs(ctx context.Context) ([]int64, error) {
 	return t.getActiveLogIDs(ctx)
 }
 
+func (t *readOnlyLogTX) GetUnsequencedCounts(ctx context.Context) (map[int64]int64, error) {
+	t.ms.mu.RLock()
+	defer t.ms.mu.RUnlock()
+
+	ret := make(map[int64]int64)
+	for id, tree := range t.ms.trees {
+		tree.mu.RLock()
+		defer tree.mu.RUnlock()
```

Comment:
Maybe add a comment, as it's difficult to guess whether the defer-in-loop is OK or a slip? It doesn't matter much in the test, as you said, but we might as well foster good practices throughout the entire codebase.

```go
defer  tree.mu.RUnlock() // OK to hold until the end of the method
```